src/libopensc/log.c

```diff
@@ -68,6 +71,26 @@ void sc_do_log_color(sc_context_t *ctx, int level, const char *file, int line, c
 	va_end(ap);
 }
 
+#ifdef ENABLE_OPENSSL
+static int log_openssl_error_line(const char *str, size_t len, void *u) {
+	struct log_info_openssl *info = (struct log_info_openssl *) u;
+	// print error message without newline character at the end
+	sc_do_log(info->ctx, info->level, info->file, info->line, info->func, "OpenSSL error: %.*s", (int) len - 2, str);
+	return 1;
```

Comment:
For example when calling `EVP_PKEY_fromdata_init(NULL)`, the OpenSSL `ERR_print_errors_cb()` function gives me error string

`00D8CDCD007F0000:error:03000096:digital envelope routines:fromdata_init:operation not supported for this keytype:crypto/evp/pmeth_gn.c:354`

That was the main reason for prepending `OpenSSL error`.

Reply:
I was wondering how this looks like in an opensc debug log. if this always has multiple long lines, then, for example, it could be more useful to have an indication on top, with multiple openssl debug log errors following

Reply:
Indeed, the resulting line in the debug log is quite long. Also, when the `sc_log_openssl` is called, the whole OpenSSL error stack is printed out, so even though it would usually be 1 error, having something like

```
P:74501; T:0x140574661335040 10:59:29.103 [opensc-tool] card-piv.c:5698:piv_init: OpenSSL error
00D8C016DA7F0000:error:03000096:digital envelope routines:fromdata_init:operation not supported for this keytype:crypto/evp/pmeth_gn.c:354
```

what is shorter and probably more readable.